model-optimizer/extensions/ops/loop.py

```diff
@@ -372,7 +375,8 @@ def re_number_output_port(loop_node: Node, old_port_id: int, new_port_id: int):
                     new_port_id += 1
 
             for port_idx_to_remove in reversed(range(new_port_id, max_port_id + 1)):
-                loop_node.delete_output_port(port_idx_to_remove)
+                if port_idx_to_remove in loop_node.out_ports().keys():
+                    loop_node.delete_output_port(port_idx_to_remove)
```

Comment:
Please, double check if this change is really needed. If you encountered a crash here then probably something was broken before.

Reply:
After body graph clean up in the transformation I have a holes in output ports so they need to be handled appropriately.